e2e/pos_test.go

```diff
@@ -499,28 +510,36 @@ func TestPoS_StakeUnstakeExploit(t *testing.T) {
 
 	// Required default values
 	numTransactions := 6
-	signer := crypto.NewEIP155Signer(100, true)
+	signer := crypto.NewSigner(chain.AllForksEnabled.At(0), 100)
 	currentNonce := 0
 
 	// TxPool client
 	txpoolClient := srv.TxnPoolOperator()
 
-	generateTx := func(value *big.Int, methodName string) *types.Transaction {
-		signedTx, signErr := signer.SignTx(&types.Transaction{
-			Nonce:    uint64(currentNonce),
-			From:     types.ZeroAddress,
-			To:       &stakingContractAddr,
-			GasPrice: bigGasPrice,
-			Gas:      framework.DefaultGasLimit,
-			Value:    value,
-			V:        big.NewInt(1), // it is necessary to encode in rlp
-			Input:    framework.MethodSig(methodName),
-		}, senderKey)
-
-		if signErr != nil {
-			t.Fatalf("Unable to sign transaction, %v", signErr)
+	generateTx := func(i int, value *big.Int, methodName string) *types.Transaction {
+		unsignedTx := &types.Transaction{
```

Comment:
Again, not sure if it is justified to adapt E2E tests that are using the IBFT codebase. I'd prefer implementing a new one from scratch but using polybft e2e framework.